kebechet/config.py

```diff
@@ -87,35 +110,87 @@ def _tls_verification(service_url: str, slug: str, verify: bool) -> None:
             return original_delete(*args, **kwargs, verify=verify)
+
         requests.Session.delete = delete
 
         def put(*args, **kwargs):
-            kwargs.pop('verify', None)
+            kwargs.pop("verify", None)
             return original_put(*args, **kwargs, verify=verify)
+
         requests.Session.put = put
 
         def get(*args, **kwargs):
-            kwargs.pop('verify', None)
+            kwargs.pop("verify", None)
             return original_get(*args, **kwargs, verify=verify)
+
         requests.Session.get = get
 
         def head(*args, **kwargs):
-            kwargs.pop('verify', None)
+            kwargs.pop("verify", None)
             return original_head(*args, **kwargs, verify=verify)
+
         requests.Session.head = head
 
         def patch(*args, **kwargs):
-            kwargs.pop('verify', None)
+            kwargs.pop("verify", None)
             return original_patch(*args, **kwargs, verify=verify)
+
         requests.Session.patch = patch
 
+    @classmethod
+    def run_analysis(cls, analysis_id: str, origin: str, service: str) -> None:
+        """Run result managers (meant to be triggered automatically)."""
+        global config
+        from kebechet.managers import ThothAdviseManager, ThothProvenanceManager
+        # Gets kebechet configuration from the original repository
+        tempfile = config.download_conf_from_url(origin, service)
+        config.from_file(tempfile.name)
+        for managers, slug, service_type, service_url, token, tls_verify in config.iter_entries():
+            cls._tls_verification(service_url, slug, verify=tls_verify)
+
+            if service_url and not service_url.startswith(("https://", "http://")):
+                # We need to have this explicitly set for IGitt and also for security reasons.
+                _LOGGER.error(
+                    "You have to specify protocol ('https://' or 'http://') in service URL "
+                    "configuration entry - invalid configuration %r",
+                    service_url,
+                )
+                continue
+
+            if (service_url and service_url.endswith('/')):
+                service_url = service_url[:-1]
+
+            if token:
+                # Allow token expansion based on env variables.
+                token = token.format(**os.environ)
+                _LOGGER.debug("Using token %r%r", token[:3], '*'*len(token[3:]))
+
+            for manager in managers:
+                manager_name = manager.pop("name")
+                if analysis_id.startswith("adviser") and manager_name == "thoth-advise":
+                    kebechet_manager = ThothAdviseManager
+                elif analysis_id.startswith("provenance") and manager_name == "thoth-provenance":
+                    kebechet_manager = ThothProvenanceManager
+                else:
+                    _LOGGER.debug("Manager %r does not correspond with id:%r, skipping", manager_name, analysis_id)
```

Comment:
I had it as a debug statement because it is expected behaviour because it should only correspond with one of the many managers defined in the config file